coresimd/src/x86/i586/sse.rs

```diff
@@ -1070,14 +1070,14 @@ pub unsafe fn _mm_loadr_ps(p: *const f32) -> f32x4 {
 // On i586 (no SSE2) it just generates plain MOV instructions.
 #[cfg_attr(all(test, any(target_arch = "x86_64", target_feature = "sse2")),
            assert_instr(movhpd))]
-pub unsafe fn _mm_storeh_pi(p: *mut u64, a: f32x4) {
+pub unsafe fn _mm_storeh_pi(p: *mut u8x8, a: f32x4) {
```

Comment:
Same here, we should make the `storeh` functions consistent with the `loadh`. That is: `p: *mut f32x2`.